tests/shardddl3/run.sh

```diff
@@ -873,6 +873,1442 @@ function DM_112() {
+}
+
+function DM_126_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1,1,1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2,2,2,2,2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3,3,3,3,3);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} drop index idx1, add index idx1(c2,c1), drop index idx2, add index idx2(c4,c3);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,4,4,4,4);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5,5,5,5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,6,6,6,6);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} drop index idx1, add index idx1(c1,c2), drop index idx2, add index idx2(c3,c4);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7,7,7,7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8,8,8,8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9,9,9,9,9);"
+
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+    run_dm_ctl_with_rematch $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "show-ddl-locks" \
+        "\"unsynced\": \[[\n ]*\]" 1
+}
+
+# Ajust multiple indexes combination and then rollback.
+function DM_126 {
+    # run_case 126 "double-source-pessimistic" \
+    # "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int primary key, c1 int, c2 int, c3 int, c4 int, index idx1(c1, c2), index idx2(c3, c4));\"; \
+    #  run_sql_source2 \"create table ${shardddl1}.${tb1} (a int primary key, c1 int, c2 int, c3 int, c4 int, index idx1(c1, c2), index idx2(c3, c4));\"; \
+    #  run_sql_source2 \"create table ${shardddl1}.${tb2} (a int primary key, c1 int, c2 int, c3 int, c4 int, index idx1(c1, c2), index idx2(c3, c4));\"" \
+    # "clean_table" "pessimistic"
+    run_case 126 "double-source-optimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int primary key, c1 int, c2 int, c3 int, c4 int, index idx1(c1, c2), index idx2(c3, c4));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int primary key, c1 int, c2 int, c3 int, c4 int, index idx1(c1, c2), index idx2(c3, c4));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int primary key, c1 int, c2 int, c3 int, c4 int, index idx1(c1, c2), index idx2(c3, c4));\"" \
+    "clean_table" "optimistic"
+}
+
+function DM_127_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2,2,2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3,3,3);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} add index idx2(c2), drop index idx1;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,4,4);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5,5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,6,6);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} add index idx1(c1), drop index idx2;"
```

Comment:
(not sure if this should be run on different tables according to our spreadsheet @GMHDBJD)